components/vbus.rst

```diff
@@ -0,0 +1,246 @@
+        collector_min:
+          name: Option Collector Min
+        collector_frost:
+          name: Option Collector Frost
+        tube_collector:
+          name: Option Tube Collector
+        recooling:
+          name: Option Recooling
+        hqm:
+          name: Option Heat Quantity Measurement
+
+
+Configuration variables:
+
+- **model** (*Mandatory*): Specify the model of the connected controller. Currently supported models are: ``deltasol_bs_plus``, ``deltasol_c``, ``deltasol_cs2``.
+
+Supported sensors:
+
+- for **deltasol_bs_plus**: ``relay1``,  ``relay2``, ``sensor1_error``, ``sensor2_error``, ``sensor3_error``, ``sensor4_error``, ``collector_max``, ``collector_min``, ``collector_frost``, ``tube_collector``, ``recooling``, ``hqm``.  
+- for **deltasol_c**: ``sensor1_error``, ``sensor2_error``, ``sensor3_error``, ``sensor4_error``.  
+- for **deltasol_cs2**: ``sensor1_error``, ``sensor2_error``, ``sensor3_error``, ``sensor4_error``.  
+
+
+All sensors are *Optional* and support all other options from :ref:`Binary Sensor <config-binary_sensor>`.
+
+
+Lambda definition for ``custom`` VBus sensors
+---------------------------------------------
+
+Devices on a VBus are identified with a source address. There can be multiple devices on the same bus, 
+each device type has a different address. 
+
+
+.. code-block:: yaml
+
+    sensor: # or binary_sensor:
+      - platform: vbus
+        model: custom
+        dest: 0x10
+        source: 0x1234
+        command: 0x100
+        lambda: |-
+          // the data is in `x`
+
+
+Configuration variables:
+
+- **model** (*Mandatory*): Set to ``custom``.  
+- **dest** (*Mandatory*): The ``DFA`` value corresponding to your device (see below).  
+- **source** (*Mandatory*): The address corresponding to ``your device model`` (see below).  
+- **command** (*Mandatory*): The ``command`` corresponding to your device (see below). 
+- **lambda** (*Mandatory*, :ref:`lambda <config-lambda>`):
+  Lambda to parse the values from the incoming data packets and publish them to template sensors.
+
+To determine the correct values for the parameters above, visit `packet definitions list <http://danielwippermann.github.io/resol-vbus/#/vsf>`__. In the search field of the **Packets** table, enter the name of your device. 
+
+To extract the values with a :ref:`lambda <config-lambda>`, look in the packet structure by clicking the **Bytes** link in the table. Each value is placed at an ``offset`` within the packet.    
+For ``float`` values, let's look at the temperatures exemple: the value is stored as ``16``-bit values in ``2`` bytes little-endian format. Since it's always the second byte containing the upper byte, it needs to be shifted by ``8`` bits (multiplied by ``256``) (e.g. ``0x34, 0x12 -> 0x1234``). The result needs to be multiplied by the factor, which is ``0.1``, to obtain the correct values: ``((x[1] << 8) + x[0]) * 0.1f)``. The number within the square brackets is the ``[offset]``.    
```

Comment:
```suggestion
For ``float`` values, let's look at the temperature example: the value is stored as a ``16``-bit value in ``2`` bytes little-endian format. Since it's always the second byte containing the upper byte, it needs to be shifted by ``8`` bits (multiplied by ``256``) (e.g. ``0x34, 0x12 -> 0x1234``). The result needs to be multiplied by the factor, which is ``0.1``, to obtain the correct values: ``((x[1] << 8) + x[0]) * 0.1f)``. The number within the square brackets is the ``[offset]``.    
```